_posts/2019-06-21-my-infatuation-with-elm.markdown

```diff
@@ -0,0 +1,76 @@
+Another feature that I haven't personally used is that it can detect outdated dependencies.
+This is helpful to keep up to date with package changes without having to check Elm Package.
+I'm thankful that this tool has allowed our code base to stay consistent and keep out minor syntactic garbage.
+
+The [Elm Package](https://package.elm-lang.org/) manager is also another aspect I enjoy about Elm because it is easy to read and understand.
+It allows you to search for packages and presents the results in a useful format.
+This allows me to find the functions I need to solve most of my needs.
+Like most package managers it links well to the related functions and types.
+I do wish that Elm Package allowed me to insert a type signature and return related functions with that type signature, like Haskell, but oh well.
+The previous two tooling points are by far my favorite, but I do appreciate a nice package manager and Elm's easy of versioning.
+Publishing a personal package is another story that I'll touch on later, but for now let us move on.
+
+## Compiler
+
+I **LOVE** Elm's compiler.
+It's instruction is that of a caring mother who just wants you to succeed even when you try and cause chaos through your code.
+Besides that it gives you a ton of confidence when I am writing new code or refactoring ancient code.
+If I asked my coworkers, I am sure that they would say it is their favorite compiler, even if most of their compiler experience is with Haskell.
+Its ability to use type inference to find all the edge cases ensures that your program does not have any runtime errors, which blows my mind.
+It also compiles quickly, which helps with development time.
+I have been impressed with the recent improvements to the compiler in the newest version.
+
+## Easy to start with an existing application
+
+Another really nice aspect of Elm is the ability to give it a shot in your existing app.
+At work we have an Angular 1.5 application that we are currently changing over to Elm piece by piece.
+This is allowing us to provide incremental changes to our users so we can get feedback quickly as we make massive improvements to our product.
+We also don't have to worry about holding out a new app until it's completed because trying to rewrite our entire app in Elm would take quite a load of resources.
+So far our leadership has really enjoyed seeing these new UI's and the frontend bugs related to our Elm elements are minimal.
+
+I would really encourage you to give this idea a try because it is as easy as creating a div element that Elm can take over and control.
+This could be a whole page or a single element, like a navigation bar.
+If it doesn't work out that's okay, at least you didn't build an entire app in a language that wasn't the right fit.
+
+## Wrapping up
+
+I love Elm, plain and simple.
+It's has great tooling, a top notch compiler, and is super easy to get started.
+I encourage you to jump in!
+Although it isn't perfect, but it is pretty close.
```

Comment:
```suggestion
Although it isn't perfect, it is pretty close.
```
You don't need both `Although` and `but` here because they're doing the same function. So keeping either one is fine, whichever you prefer.